speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/ResearcherMenu.cpp

```diff
@@ -306,6 +332,30 @@ static void LoadConfigSettings(void* p_param)
     // Set the max time setting from the xml file
     m_maxTime = std::stoi(GfParmGetStr(p_param, PRM_MAX_TIME, GFMNU_ATTR_TEXT, NULL));
 
+    const char* filePath = GfParmGetStr(p_param, PRM_BLACKBOX, "name", NULL);
+    if (filePath != nullptr) 
+    {
+        strcpy_s(m_blackBoxFilePath, BLACKBOX_PATH_SIZE, filePath);
+        SMediator* mediator = SMediator::GetInstance();
+        mediator->SetBlackBoxFilePath(m_blackBoxFilePath);
+        m_blackBoxChosen = true;
+        std::string fileName;
+        fileName.resize(BLACKBOX_PATH_SIZE);
+        for (int i = 0; i < BLACKBOX_PATH_SIZE; i++)
+        {
+            fileName[i] = m_blackBoxFilePath[i];
+        }
+        int lastDirectoryIndex = 0;
+        for (int i = fileName.size() - 1; i >= 0; i--)
+        {
+            if (fileName[i] != '\\') { continue; }
+            lastDirectoryIndex = i;
+            break;
+        }
```

Comment:
This seems like something that could be placed in a helper function GetFileNameFromPath() or something like that